src/main/java/io/github/thebusybiscuit/slimefun4/utils/SlimefunUtils.java

```diff
@@ -293,13 +294,19 @@ private static boolean equalsItemMeta(@Nonnull ItemMeta itemMeta, @Nonnull ItemM
             return false;
         } else if (itemMeta.hasDisplayName() && sfitemMeta.hasDisplayName() && !itemMeta.getDisplayName().equals(sfitemMeta.getDisplayName())) {
             return false;
-        } else if (!checkLore) {
-            return true;
-        } else if (itemMeta.hasLore() && sfitemMeta.hasLore()) {
-            return equalsLore(itemMeta.getLore(), sfitemMeta.getLore());
-        } else {
-            return !itemMeta.hasLore() && !sfitemMeta.hasLore();
+        } else if (checkLore) {
+            if (itemMeta.hasLore() && sfitemMeta.hasLore() && !equalsLore(itemMeta.getLore(), sfitemMeta.getLore())) {
+                return false;
+            } else if (itemMeta.hasLore() != sfitemMeta.hasLore()) {
+                return false;
+            }
```

Comment:
I think we should make variables here for `hasItemMetaLore` and `hasSfItemMetaLore`.